ambry-rest/src/main/java/com/github/ambry/rest/AsyncRequestResponseHandler.java

```diff
@@ -299,10 +299,18 @@ public void run() {
           } else {
             break;
           }
-        } catch (Exception e) {
+        } catch (Throwable e) {
+          if (!(e instanceof Exception)) {
+            metrics.requestProcessingThrowableError.inc();
+            if (requestInfo != null) {
+              logger.error("Throwable only while processing the requests" + requestInfo.restRequest, e);
+            } else {
+              logger.error("Throwable only while processing the requests", e);
+            }
+          }
           metrics.requestProcessingError.inc();
           if (requestInfo != null) {
-            onProcessingFailure(requestInfo.restRequest, requestInfo.restResponseChannel, e);
+            onProcessingFailure(requestInfo.restRequest, requestInfo.restResponseChannel, (Exception) e);
```

Comment:
you can't convert this to an Exception if e is not an instance of Exception. you can probably add a line in the previous if statement to convert this throwable to exception. something like this 
```
  if (!(e instanceof Exception)) {
            metrics.requestProcessingThrowableError.inc();
            if (requestInfo != null) {
              logger.error("Throwable only while processing the requests" + requestInfo.restRequest, e);
            } else {
              logger.error("Throwable only while processing the requests", e);
            }
           e = new Exception(e);
          }
```

Reply:
you are right, fixed it.